hanging_threads.py

```diff
@@ -169,7 +169,13 @@ def frame2string(frame):
     filename = co.co_filename
     name = co.co_name
     s = '\tFile "{0}", line {1}, in {2}'.format(filename, lineno, name)
-    line = linecache.getline(filename, lineno, frame.f_globals).lstrip()
+    try:
+        line = linecache.getline(filename, lineno, frame.f_globals).lstrip()
```

Comment:
I would assume this is due to `lineno == None` which can be tested before the call of the function.

Reply:
I think so; but there was one other component that I was not very sure of. Therefore I thought try-catch would be a better patch here.